File: cpp/flashmen/night_test/11_13.py
n, m, x = map(int, input().split())
qwq = []
cost = [0 for i in range(n)]
skill = [0 for i in range(m)]
for i in range(n):
    tmp = list(map(int, input().split()))
    cost[i] = tmp[0]
    qwq.append(tmp[1:])
ans = 1000000000

res = 0


def check(qwq):
    for i in qwq:
        if i < x:
            return False
    return True


def dfs(i):
    global skill, res, ans
    if i == n:
        if check(skill):
            ans = min(ans, res)
        return
    if not (check(skill)):
        res += cost[i]
        skill = [qwq[i][j] + skill[j] for j in range(m)]
        dfs(i + 1)
        skill = [skill[j] - qwq[i][j] for j in range(m)]
        res -= cost[i]
    dfs(i + 1)


dfs(0)
print(ans if ans != 1000000000 else -1)

File: cpp/flashmen/night_test/11_13.cpp
// Copyright 2024 Allergy
/* @author    Allergy
 * @email     Allergy527@gmail.com
 * @workspace cpp\flashmen\night_test\11_13.cpp
 * @date      2024/11/13 18:57:41
 */
#include <bits/stdc++.h>
#define lowbit(x) ((x) & (-(x)))
#define ln '\n'
#define int int64_t

using namespace std;  // NOLINT
using ll = int64_t;
int inf = 0x3f3f3f3f;
ll ll_inf = 0x3f3f3f3f3f3f3f3f;

void solve() {
    int n, m, x;
    cin >> n >> m >> x;
    vector<vector<int>> qwq(n, vector<int>(m));
    vector<int> cost(n);
    for (int i = 0; i < n; ++i) {
        cin >> cost[i];
        for (auto &x : qwq[i]) cin >> x;
    }
    vector<int> skill(m);
    auto check = [&]() {
        for (auto t : skill)
            if (t < x) return false;
        return true;
    };
    int ans = inf;
    int res = 0;
    auto dfs = [&](auto self, int i) {
        if (i == n) {
            if (check()) ans = min(ans, res);
            return;
        }
        if (!check()) {
            res += cost[i];
            for (int j = 0; j < m; ++j) skill[j] += qwq[i][j];
            self(self, i + 1);
            for (int j = 0; j < m; ++j) skill[j] -= qwq[i][j];
            res -= cost[i];
        }
        self(self, i + 1);
    };
    dfs(dfs, 0);
    cout << (ans == inf ? -1 : ans) << ln;
}
signed main() {
    ios::sync_with_stdio(false), cin.tie(0), cout.tie(0);
    int t = 1;
    // cin >> t;
    while (t--) solve();
    cin >> t;
    return 0;
}
